test/dialing/direct.node.ts

```diff
@@ -180,7 +180,9 @@ describe('Dialing (direct, TCP)', () => {
       .and.to.have.property('code', ErrorCodes.ERR_TIMEOUT)
   })
 
-  it('should dial to the max concurrency', async () => {
+  // The implementation of DialRequest is not capable of preventing more dial attempts
+  // than allowed by `maxParallelDials`
+  it.skip('should dial to the max concurrency', async () => {
```

Comment:
We should only ever dial to the max concurrency, this test should still pass with these changes - can you please fix it up instead of skipping it?

Reply:
This is supposed to be a non-trivial change. In the current version, the test runs through because the bug is covered by this issue.

I would rather fix this in a follow-up PR to keep this PR simple.

Reply:
Changed - turned out to be easier than expected

Reply:
Awesome, I had a feeling it might have been 😉 

Please can you re-enable this test and then I think this is good to go.

Reply:
changeg